velox/dwio/parquet/tests/reader/ParquetTableScanTest.cpp

```diff
@@ -674,6 +703,43 @@ TEST_F(ParquetTableScanTest, sessionTimezone) {
   assertSelectWithTimezone({"a"}, "SELECT a FROM tmp", "Asia/Shanghai");
 }
 
+TEST_F(ParquetTableScanTest, remainingFilterLazyWithMultiReferences) {
+  constexpr int kSize = 10;
+  auto vector = makeRowVector({
+      makeFlatVector<int64_t>(kSize, folly::identity),
+      makeFlatVector<int64_t>(kSize, folly::identity),
+      makeFlatVector<int64_t>(kSize, folly::identity),
+  });
+  auto schema = asRowType(vector->type());
+  auto file = TempFilePath::create();
+  writeToFile(file->getPath(), {vector});
+  CursorParameters params;
+  params.copyResult = false;
+  params.singleThreaded = true;
+  params.planNode =
+      PlanBuilder()
+          .tableScan(schema, {}, "NOT (c0 % 2 == 0 AND c2 % 3 == 0)")
```

Comment:
You don't need to do such fancy test, just read one column untouched and check it's lazy encoded, then load it and validate the content.

Reply:
I modified `NOT (C0% 2 == 0 AND c2%3 == 0)` to `c0 >= 0` to read out all the data, where `c1` and` c2` should be loaded via `loadedVector`.